### PR TITLE
Add provider equinix and module_name in Equinix Metal TF configs

### PR DESCRIPTION
**Pull request by codinja1188**

### CODE REVIEW

File: example/versions.tf [hunk 1]
```diff
@@ -3,7 +3,10 @@ terraform {
   required_providers {
     equinix = {
       source  = "equinix/equinix"
-      version = "= 1.1.0"
+      version = "~> 1.14"
     }
   }
+  provider_meta "equinix" {
```
Comment (ctreatma) [inline example/versions.tf]: @ocobleseqx suggested that we should not include `provider_meta` in example modules, and I'm on board with that; let's remove this.

**Review by ctreatma — CHANGES_REQUESTED** [review]
There are a couple of CI failures that need to be addressed

File: example/versions.tf (hunk 1 above)
Comment (codinja1188) [reply example/versions.tf]: updated

**Comment by ctreatma** [comment]
The CI job that runs the example is still failing: https://github.com/equinix/terraform-equinix-pa-vm/actions/runs/4950445319/jobs/8854067051?pr=3

If there are `.terraform.lock.hcl` files in the repo, probably better to `git rm` them and add `.terraform.lock.hcl` to the `.gitignore`.

**Comment by ctreatma** [comment]
@vasubabu It looks like `example/main.tf` is using the published version of this module, not using the local copy.  I think if you update the `source` for the module in that file, it will fix the failing CI job

File: example/main.tf [hunk 2]
```diff
@@ -5,7 +5,6 @@ provider "equinix" {
 
 module "pa-vm" {
   source               = "equinix/pa-vm/equinix"
```
Comment (ctreatma) [inline example/main.tf]: This `source` needs to be changed to point to the root module; we should also include commented-out lines that show how to pull down the published module.  Here's an example of what that looks like in the CPEM example of the `equinix-labs/terraform-equinix-kubernetes-cluster` module: https://github.com/equinix-labs/terraform-equinix-kubernetes-cluster/blob/main/examples/cpem-add-on/main.tf#L2C3-L4

File: example/main.tf [hunk 3]
```diff
@@ -4,7 +4,7 @@ provider "equinix" {
 }
 
 module "pa-vm" {
-  source               = "equinix/pa-vm/equinix"
+  source               = ".././"
   version              = "1.0.0"
```
Comment (ctreatma) [inline example/main.tf]:
```suggestion
  source               = "../../"
  # source               = "equinix/pa-vm/equinix"
  # version              = "1.1.0" # Use the latest version, according to https://github.com/equinix/terraform-equinix-pa-vm/releases
```

File: example/main.tf [hunk 4]
```diff
@@ -4,7 +4,7 @@ provider "equinix" {
 }
 
 module "pa-vm" {
-  source               = "equinix/pa-vm/equinix"
+  source               = ".././"
```
Comment (displague) [inline example/main.tf]: This should work:
```suggestion
  source               = "../"
```

File: example/main.tf [hunk 5]
```diff
@@ -4,8 +4,9 @@ provider "equinix" {
 }
 
 module "pa-vm" {
-  source               = "equinix/pa-vm/equinix"
-  version              = "1.0.0"
+  source               = "../../"
```
Comment (ctreatma) [inline example/main.tf]:
```suggestion
  source               = "../"
```

File: example/main.tf [hunk 6]
```diff
@@ -4,8 +4,9 @@ provider "equinix" {
 }
 
 module "pa-vm" {
-  source               = "equinix/pa-vm/equinix"
-  version              = "1.0.0"
+  source               = "../"
```
Comment (ctreatma) [inline example/main.tf]:
```suggestion
  source = "../"
```

**Review by ctreatma — APPROVED** [review]

**Pull request merged**